### PR TITLE
Remove "ember-cli" dependency from "resource" blueprint

### LINKED ISSUE #11: "resource" blueprint depends on missing "ember-cli" dependency

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves #11 

I'm not sure if the black magic that I've used is such a good idea, but I'm open to alternatives.

/cc @stefanpenner @trabus @rwjblue

### CODE REVIEW

File: blueprints/resource/index.js [hunk 1]
```diff
@@ -15,9 +14,11 @@ module.exports = {
   uninstall: function(options) {
     return this._process('uninstall', options);
   },
-  
+
   _processBlueprint: function(type, name, options) {
     var mainBlueprint = this.lookupBlueprint(name);
+
+    var that = this;
```
Comment (trabus) [inline blueprints/resource/index.js]: Personally, I'd prefer using bind:
```
     return Promise.resolve()
        .then(function() {
          return mainBlueprint[type](options);
       })
       .then(function() {
         var testBlueprint = mainBlueprint.lookupBlueprint(name + '-test', {
           ui: this.ui,
           analytics: this.analytics,
           project: this.project,
           ignoreMissing: true
         });
  
          if (!testBlueprint) { return; }
  
         var Blueprint = this._findBlueprintBaseClass(testBlueprint);
         if (Blueprint && testBlueprint.locals === Blueprint.prototype.locals) {
            testBlueprint.locals = function(options) {
              return mainBlueprint.locals(options);
            };
         }
 
         return testBlueprint[type](options);
        }.bind(this));
```

Comment (Turbo87) [inline blueprints/resource/index.js]: yeah, but then the existing mentions of `this` in the code would be differently bound and I wasn't sure if that would work as well

**Comment by trabus** [comment]
I'm fine with this. :+1:  @stefanpenner ?

**Comment by stefanpenner** [comment]
:+1:

**Comment by Turbo87** [comment]
Just keep in mind that we likely need to revisit this once the new core-object is merged into ember-cli

**Pull request merged**